### ISSUE TITLE
Images in XML sitemap are always linked to base store in multistore on Schedule 

### ISSUE THREAD

**Issue by lano-vargas** [magento/magento2#19596]
Website URL 1: https://my_website.com
Website URL 2: https://my_website.co.uk
### Preconditions (*)

1. Magento2.2.2 
2. Have 2 websites, 2 store and 2stores view. 1 store and 1 store view per website
3. Each website/store view  with different domain
4. A set of Configurable/simple.
2.Cron job running

### Steps to reproduce (*)

1. Got to Stores->Configuration->Catalog->XML Sitemap->General Settings:
2.Set to Enable, Start Time, Frequency Daily etc... 
**TIP:** I've set to a few minutes before checking!

### Expected result (*)

1. Website 1 should have images url referencing Website 1 domain = **https://my_website.com**
2.
```
<image:image>
<image:loc>
https://my_website.com/media/catalog/product/cache/c9e0b0ef589f3508e5ba515cde53c5ff/t/t/test1.jpg
</image:loc>
<image:title>TEST</image:title>
<image:caption>TEST</image:caption>
</image:image>
```

### Actual result (*)

1. Website 1 have images url referencing Website 2 domain = **https://my_website.nl**
2.
```
<image:image>
<image:loc>
https://my_website.nl/media/catalog/product/cache/c9e0b0ef589f3508e5ba515cde53c5ff/t/t/test1.jpg
</image:loc>
<image:title>TEST</image:title>
<image:caption>TEST</image:caption>
</image:image>
```

I've applied the fix https://github.com/magento/magento2/commit/32c2a19ddc6c56505eaf72f90b198cc1b1497023

specified here:https://github.com/magento/magento2/issues/15588

For manual XML generation works fine But when the Schedule runs next time again it add the bug back.

**Comment by magento-engcom-team** [comment]
Hi @lano-vargas. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@lano-vargas do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96989`, `MAGETWO-96990` were created

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by Nazar65** [comment]
I'm will work on this

**Comment by richardmcauleywed2b** [comment]
Hi @Nazar65 - Thanks for looking into this issue. Has there been much progress on resolving the issue at all?

**Comment by Nazar65** [comment]
Hi @richardmcauleywed2b yes this is resolved in my PR, and will available on 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @lano-vargas. Thank you for your report. 
The issue has been fixed in magento/magento2#19598 by @Nazar65 in 2.3-develop branch
Related commit(s):
 - [d5b0ffdd32ccc0d6ed23b41b33617958a659fc75](https://github.com/magento/magento2/commit/d5b0ffdd32ccc0d6ed23b41b33617958a659fc75)
 - [16f11f760b89bb240ce2f6948214da5775695d92](https://github.com/magento/magento2/commit/16f11f760b89bb240ce2f6948214da5775695d92)
 - [c8afe9f71d99367bcdd24e4900e60bbba2fbc0dd](https://github.com/magento/magento2/commit/c8afe9f71d99367bcdd24e4900e60bbba2fbc0dd)
 - [fda07a290af211356db74eb0cb890dc4bf42a388](https://github.com/magento/magento2/commit/fda07a290af211356db74eb0cb890dc4bf42a388)
 - [5c7527e47e5d983c7faccabafef1a8e3fb68d27d](https://github.com/magento/magento2/commit/5c7527e47e5d983c7faccabafef1a8e3fb68d27d)
 - [d2d04b6ee9df35b9906f16459bdee1bba0db658d](https://github.com/magento/magento2/commit/d2d04b6ee9df35b9906f16459bdee1bba0db658d)
 - [b88fa48da750a1f7be759b37d9007bf2ecdc5270](https://github.com/magento/magento2/commit/b88fa48da750a1f7be759b37d9007bf2ecdc5270)
 - [85a2a561bf5f5fe585c27ef40564692860f31be3](https://github.com/magento/magento2/commit/85a2a561bf5f5fe585c27ef40564692860f31be3)
 - [a04cbf2acd3a149d9349ec2ea738010190dc7e10](https://github.com/magento/magento2/commit/a04cbf2acd3a149d9349ec2ea738010190dc7e10)

The fix will be available with the upcoming  release.
